File: Python/Get_Data/Single_transaction_query.py
##
# BluePay Ruby Sample code.
#
# This code sample runs a report that grabs a single transaction
# from the BluePay gateway based on certain criteria.
# See comments below on the details of the report.
# If using TEST mode, only TEST transactions will be returned.
##
from __future__ import print_function
import os.path, sys
sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), os.pardir))
from BluePay import BluePay

account_id = "Merchant's Account ID Here"
secret_key = "Merchant's Secret Key Here"
mode = "TEST"  

report = BluePay(
    account_id = account_id, 
    secret_key = secret_key, 
    mode = mode 
)

report.get_single_trans_query(
    transaction_id = '100532208235', # Transaction ID
    report_start = "2015-01-01", # Query Start Date: Jan. 1, 2013
    report_end = "2018-12-31", # Query End Date: Jan. 15, 2015
    exclude_errors =  "1" # Do not include errored transactions? Yes
 ) 

# Makes the API Request with BluePay
report.process()

# Reads the response from BluePay
print(report.response)

File: Perl/Get_Data/Single_Transaction_Query.pl
##
# BluePay Perl Sample code.
#
# This code sample runs a report that grabs a single transaction
# from the BluePay gateway based on certain criteria.
# See comments below on the details of the report.
# If using TEST mode, only TEST transactions will be returned.
##

use strict;
use lib '..';
use bluepay;

my $account_id = "Merchant's Account ID Here";
my $secret_key = "Merchant's Secret Key Here";
my $mode = "TEST";

my $report = BluePay->new(
	$account_id, 
	$secret_key, 
	$mode
);

$report->get_single_transaction_query({
	transaction_id => 'ID of previous transaction', # required
	report_start_date => '2013-01-01', #YYYY-MM-DD, required
	report_end_date => '2015-05-30', #YYYY-MM-DD, required
	exclude_errors => '1' # Do not include errored transactions? Yes
});

# Makes the API request with BluePay        
$report->process();

if ($report->{payment_type}) {
# Get response from BluePay
print "TRANSACTION ID: " . $report->{id} . "\n";
print "FIRST NAME: " . $report->{name1} . "\n";
print "LAST NAME: " . $report->{name2} . "\n";
print "PAYMENT TYPE: " . $report->{payment_type} . "\n";
print "TRANSACTION TYPE: " . $report->{trans_type} . "\n";
print "AMOUNT: " . $report->{amount} . "\n";
print "CARD TYPE: " . $report->{payment_account} . "\n";
} else {
	print $report->{response};
}

